player/lua/ytdl_hook.lua

```diff
@@ -938,6 +1013,7 @@ function run_ytdl_hook(url)
 
             -- can't change the http headers for each entry, so use the 1st
             set_http_headers(json.entries[1].http_headers)
+            set_cookies(json.entries[1].cookies or json.cookies, false)
```

Comment:
Sorry this one isn't totally clear to me. When would this be used and why would we not just append it to `stream-lavf-o` normally like we do for the other call of this function?

Reply:
Sorry, I'm not sure I understand. Are you talking about the invocation of `set_cookies()` or about the function itself? yt-dlp removes cookies from `http_headers`, hence the added invocation. The second argument being `false` is consistent with the `file-local-options/...` prefix in `set_http_headers()`.

As for the function itself, do you want me not to use `mp.set_property_native`? I just tried `mp.set_property(option_key .. "-set", "cookies=[" .. lavf_cookies .. "]")` and `mp.set_property(option_key .. "-add", "cookies=" .. edl_escape(lavf_cookies))` and `mp.set_property(option_key .. "-add", "cookies=" .. lavf_cookies)`, but none of them work, and I'm not very familiar with mpv scripting.

Reply:
I meant this particular function call since it's not needed to play the example link you gave earlier (makes sense, this part of the code is for playlists). I was just wondering if you had an example that would actually execute this code and if the extra `false` argument here even makes sense. Sure it matches what `set_http_headers` does, but I just wanted to make sure that's actually correct.

Reply:
I couldn't find a suitable website, so I made my own. Try: `mpv -v ytdl://https://handlerug.me/mpv-ytdl-cookies-test`. This specific URL sets a cookie, and nginx is configured to respond with 403 for every other URL if the cookie is not present. It also redirects to an HTML file that contains two .m3u8 URLs formatted as Open Graph (yes), which in turn point to a bunch of .ts files. mpv will fail without the line you left a comment on.

Reply:
Nice, thanks for coming up with a test. This URL works for me regardless if I use `true` or `false` in the `set_cookies` call. Maybe just simplify the function to always be the same as the true behavior? Unless you can think of some reason why it wouldn't work.

Reply:
Done